src/treecons.ml

```diff
@@ -0,0 +1,131 @@
+(** Tree constructors: a simpler representation of terms to be used by a
+    decision tree. *)
+
+open Extra
+
+(** A constructor is the representation of a symbol along with the number of
+    arguments to which it is applied. *)
+type treecons =
+  | Symb of
+      { c_mod : string list
+      (** Module name where the symbol of the constructor is defined. *)
+      ; c_sym : string
+      (** Symbol of the constructor. *)
+      ; c_ari : int
+      (** Arity of the considered constructor.  A same symbol representation
+          may generate several constructors with different arities. *) }
+  | Abst
+  (** An abstraction*)
+  | Vari of string
+  (** A free variable with a name. *)
```

Comment:
What do you mean by free here? Does it correspond to a pattern variable or a variable bound by a lambda?

Reply:
A variable bound by a lambda

Reply:
Then, it's not a "free" variable but a "bound" one! ;-)